app/src/main/java/com/codingblocks/chatter/DashboardActivity.java

```diff
@@ -60,9 +62,7 @@ protected void onCreate(Bundle savedInstanceState) {
                 avatarUrl.equals("")) {
             if (isNetworkAvailable()) {
                 Request request = new Request.Builder()
-                        .url("https://gitter.im/v1/user/me")
-                        .addHeader("Accept", "application/json")
-                        .addHeader("Authorization:", "Bearer " + accessToken)
+                        .url("https://api.gitter.im/v1/user?access_token=" + accessToken)
```

Comment:
Don;t do this, it can lead to a serious security vulnerability. Don't transmit the accessToken over the url in the form of get!

Reply:
Actually i adopted the method given by gitter api guidelines

Reply:
But it's a security risk. As you know what you are doing is posting the token over as a GET request parameter. Even with https, anybody on the network can read urls to which data is flowing to and from and that token can be read easily.

Reply:
I know that but you can't change the api documentation you have to follow that right?
 https://developer.gitter.im/docs/user-resource give it a read

Reply:
Yea I've read that:
`curl -i -H "Accept: application/json" -H "Authorization: Bearer {{token}}" "https://api.gitter.im/v1/user"`

Can;t you enumerate it? It works via headers as well!

Reply:
Okay 😅 will do that thanks for the heads up

Reply:
^^